packages/opentelemetry-resources/src/platform/node/detectors/AwsBeanstalkDetector.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Resource } from '../../../Resource';
+import { SERVICE_RESOURCE } from '../../../constants';
+import { Detector } from '../../../types';
+import { ResourceDetectionConfigWithLogger } from '../../../config';
+import * as fs from 'fs';
+import * as util from 'util';
+
+/**
+ * The AwsBeanstalkDetector can be used to detect if a process is running in AWS Elastic
+ * Beanstalk and return a {@link Resource} populated with data about the beanstalk
+ * plugins of AWS X-Ray. Returns an empty Resource if detection fails.
+ *
+ * See https://docs.amazonaws.cn/en_us/xray/latest/devguide/xray-guide.pdf
+ * for more details about detecting information of Elastic Beanstalk plugins
+ */
+class AwsBeanstalkDetector implements Detector {
+  readonly BEANSTALK_CONF_PATH = '/var/elasticbeanstalk/xray/environment.conf';
```

Comment:
I'm not a beanstalk expert, but is it guaranteed to be a linux environment? Also, think about using the node `path` module to normalize this path and prevent runtime exceptions being thrown.

Reply:
I am also not so sure here, I have read through the Java code: [Java implementation](https://github.com/open-telemetry/opentelemetry-java/blob/a59748a904e0cb1e2e4f8df50dd346a82f26ff1e/sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/BeanstalkResource.java). And found they seem to use this path for sure. Also in [AWS X-Ray dev doc](https://docs.aws.amazon.com/xray/latest/devguide/xray-guide.pdf), they does not mention whether there is difference between different OS and PL.
As for checking file existence, I also think is a good idea, will add it!
By the way, it seems that when promisify the ```fs.readFile``` outside of the class, ```sinon.sandbox.stub``` cannot simply stub readFile function by ```sandbox.stub(fs, 'readFile').yields(null, JSON.stringify(data));``` anymore. Any suggestions for that?

Reply:
In addition, in [aws-xray-sdk-node](https://github.com/aws/aws-xray-sdk-node/blob/695f0498d3455a98c393d3720ee213384527f115/packages/core/lib/segments/plugins/elastic_beanstalk_plugin.js) They also take this for sure.

Reply:
This is an interesting point, I wouldn't be surprised if this code predates Windows Beanstalk. @shengxil do you happen to have a pointer on how to read beanstalk env config?

Reply:
What is the status here? As I said above I'm not a beanstalk expert so I don't want to just approve and merge this if there is a chance it will cause failures down the line.

Reply:
As for this issue, I got responded from our experienced AWS engineer: "Beanstalk manages all its own platforms, and they do run Windows, however it should be the same".
In [AWS X-Ray developer documentation](https://docs.aws.amazon.com/xray/latest/devguide/xray-guide.pdf), it explicitly points out the config path is ```/var/elasticbeanstalk/xray/environment.conf```
And I checked [Beanstalk documentation](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/environment-configuration-debugging.html#environment-configuration-debugging-namespace), the namespace ```aws:elasticbeanstalk:xray``` corresponds to ```/var/elasticbeanstalk/xray/environment.conf```.
Although in Beanstalk documentation, they did not explicitly point the config path out, the [aws-xray-sdk-node](https://github.com/aws/aws-xray-sdk-node/blob/695f0498d3455a98c393d3720ee213384527f115/packages/core/lib/segments/plugins/elastic_beanstalk_plugin.js) and [aws-xray-sdk-ruby-doc](https://docs.aws.amazon.com/xray-sdk-for-ruby/latest/reference/XRay/Plugins/ElasticBeanstalk.html) both support this.
In conclusion, I think directly using this config path is right here.

Reply:
UPDATE: Seen from [aws-xray-sdk-dotnet](https://github.com/aws/aws-xray-sdk-dotnet/blob/8053d17f191d40cbabb18b4cb20b6d6ff6cfb44f/sdk/src/Core/Plugins/ElasticBeanstalkPlugin.cs#L32), we now know the path of windows OS should be ```'C:\\Program Files\\Amazon\\XRay\\environment.conf'```, will add this part and make another commit!